brave/brave-core
# Added ms-edge protocol handler option

fix brave/brave-browser#17558

Applied below FF's implementation to set default handler w/o using windows system ui.
It calculated hash value by using some properties such as user sid, protocol or timestamp.
If this method is not available, system ui will be launched instead.
https://github.com/mozilla/gecko-dev/blob/master/toolkit/mozapps/defaultagent/SetDefaultBrowser.cpp


Resolves 

Option state when brave is not yet set for ms-edge protocol handler
![Screenshot 2021-11-04 212457](https://user-images.githubusercontent.com/6786187/140313348-47413938-84df-43be-8b33-153b72388dea.png)

Option state when brave is ms-edge protocol handler
![Screenshot 2021-11-04 212648](https://user-images.githubusercontent.com/6786187/140313352-32924b20-ec5a-4087-801f-21062a9b0102.png)

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
`npm run test brave_unit_tests -- --filter=DefaultProtocolHandlerUtilsWinTest`

### Test 1
1. Check this browser is not a default ms-edge protocol handler from control panel
2. Launch browser and load `brave://settings/system/`
3. Check `Make default`  button is visible and click
4. Check button is disappeared
5. Search any keyword from windows taskbar and bing url is loaded in this browser
6. Change default handler to another browser like edge from control panel
7. Check `brave://settings/system/` is updated properly

### Test 2
1. Check this browser is not a default ms-edge protocol handler from control panel
2. Launch browser with ` --use-system-ui-for-ms-edge` cmd line switch
3.  load `brave://settings/system/` and click button
4. Check system ui is launched for setting default apps for protocol

NOTE: With Older than Windows 10 RS2, system ui will be launched.
